.github/CONTRIBUTING.md

```diff
@@ -0,0 +1,187 @@
+# Contributing
+
+We would love for you to contribute and help make it even better
+than it is today! As a contributor, here are the guidelines we would like you
+to follow:
+
+ - [Question or Problem?](#question)
+ - [Issues and Bugs](#issue)
+ - [Feature Requests](#feature)
+ - [Submission Guidelines](#submit)
+ - [Coding Rules](#rules)
+ - [Commit Message Guidelines](#commit)
+
+## <a name="question"></a> Got a Question or Problem?
+
+Please, do not open issues for the general support questions as we want to keep GitHub issues for bug reports and feature requests. 
+
+## <a name="issue"></a> Found an Issue?
+If you find a bug in the source code or a mistake in the documentation, you can help us by
+[submitting an issue](#submit-issue) to our [GitHub Repository][github]. Even better, you can
+[submit a Pull Request](#submit-pr) with a fix.
+
+## <a name="feature"></a> Want a Feature?
+You can *request* a new feature by [submitting an issue](#submit-issue) to our [GitHub
+Repository][github]. If you would like to *implement* a new feature, please submit an issue with
+a proposal for your work first, to be sure that we can use it.
+
+First open an issue and outline your proposal so that it can be
+discussed. This will also allow us to better coordinate our efforts, prevent duplication of work,
+and help you to craft the change so that it is successfully accepted into the project.
+
+**All features require a proper design and review by team members and product owners.** Before starting work, you might want to
+discuss with us to figure out:
+
+* Is this something we want?
+* What's the impact?
+
+Answering those questions first in the request might help us make a decision.
+
+## <a name="submit"></a> Submission Guidelines
+
+### <a name="submit-issue"></a> Submitting an Issue
+
+Before you submit an issue, please search the issue tracker, maybe an issue for your problem already exists and the discussion might inform you.
+
+We want to fix all the issues as soon as possible, but before fixing a bug we need to reproduce and confirm it. Having a reproducible 
+scenario gives us wealth of important information without going back & forth to you with additional questions.
```

Comment:
Maybe add suggestion to create, and link to, a repo that includes minimal code to reproduce the problem.

Reply:
I added `If you can, please provide a repository with minimal reproduction steps.`